### DISCUSSION TITLE
Making software development an integral part of infectious disease modelling projects

### DISCUSSION THREAD

**Discussion by JUNIPER-partnership** [epiverse-trace/epiverse-trace.github.io#343] [General]
I had a question about how we make sure that software development is an integral part of infectious disease modelling projects. Is the way forward that we need to have proper software packages for the individual building blocks of infectious disease models, and then train modellers how to use these packages? Do we need to be trying to change the funding landscape so that funded research projects have an research software engineer (RSE) in the same way they would have a post-doctoral researcher (PDRA)? Or should it be some combination of the two? Interested to hear people's thoughts.

**Comment by adamkucharski** [comment]
Good question! Have been reflecting on this quite a bit post-COVID. I suppose it partly comes down to 'what is our role in a modelling project?' There are (at least) four roles involved in modelling work:
1. Developing methodology
2. Implementing methodology as local prototype code (functions, scripts)
3. Applying code to a scientific/public health question
4. Making stable code available for wider audience to use (documented packages, libraries)

Traditionally, modelling research has mostly spanned 1-3, with (4) less common. And while training (e.g. during MSc/PhD projects), people may focus just on a smaller subset. But each of these require specific skills, so it's worth considering how we can do them more efficiently, e.g. (4) really benefits from the expertise of an RSE (or extremely technically-minded researcher) if we want a package to be genuinely useable and stable. (Increasingy, if I see 'develop a package' as a deliverable in a grant application without software engineering experience or support, I flag it as a risk to success.) Also, having students routinely spend a lot of time on (2) probably isn't great if they're doing an applied MSc or PhD (just as stats students don't necessarily reimplement `mgcv` and  phylogenetics  students don't reimplement BEAST for analysis). It's been nice to see students now able to take on more ambitious scientific problems with tools like odin/dust and Epiverse packages available.

But keen to hear others' thoughts!
